rgbd/include/rgbd/Client.h

```diff
@@ -1,40 +1,38 @@
 /**
- * This client can either be used to listen to RGBD messages on a single topic
- * (use initialize(_)) or to seperate RGB / Depth / camera info topics
- * (use initialize(_, _, _)).
- *
- * The client will always convert the incoming data to rgbd::Image data structures.
+ * This client listens to RGBD messages on a single topic or via shared memory, directy rgbd::Image.
  */
 
-#ifndef RGBD_TRANSPORT_CLIENT_H_
-#define RGBD_TRANSPORT_CLIENT_H_
+#ifndef RGBD_CLIENT_H_
+#define RGBD_CLIENT_H_
 
-#include <ros/ros.h>
-
-#include "rgbd_msgs/RGBD.h"
+#include <ros/node_handle.h>
+#include <ros/subscriber.h>
 #include <ros/callback_queue.h>
 
 #include "rgbd/types.h"
-#include "rgbd/shared_mem_client.h"
+#include "rgbd/Client_RGBD.h"
+#include "rgbd/Client_SHM.h"
 
+#include "rgbd_msgs/RGBD.h"
 
-// ROS message serialization
-#include <sensor_msgs/Image.h>
-#include <sensor_msgs/CameraInfo.h>
 
 namespace rgbd {
 
-// Forward declare struct that holds all approximate time sync stuff
-struct ROSImageSyncData;
-
-class Image;
-
+/**
```

Comment:
So nice that there are docstrings everywhere so that at least we have an idea of what's going on ;-)